### ISSUE TITLE
`invalid-name` type alias false-positive for union variables without assignment values

### ISSUE THREAD

**Issue by yilei** [pylint-dev/pylint#8540]
### Bug description

```
# pylint: disable=missing-docstring
from typing import Union
x: Union[str, int]
```

### Command used

```shell
pylint t.py
```

### Pylint output

```shell
************* Module t
t.py:3:0: C0103: Type alias name "x" doesn't conform to predefined naming style (invalid-name)
```

### Expected behavior

No output.

### Pylint version

```shell
pylint 2.17.2
astroid 2.15.2
Python 3.11.1 (main, Jan 10 2023, 15:08:01) [Clang 14.0.0 (clang-1400.0.29.202)]
```

### Additional dependencies

Similar to #8487. This is a bug found in #8537, let me send a separate PR first.
